blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DBasicMetricsRW.java

```diff
@@ -0,0 +1,350 @@
+package com.rackspacecloud.blueflood.io.datastax;
+
+import com.codahale.metrics.Timer;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.ResultSetFuture;
+import com.datastax.driver.core.Row;
+import com.google.common.annotations.VisibleForTesting;
+import com.netflix.astyanax.serializers.StringSerializer;
+import com.rackspacecloud.blueflood.exceptions.CacheException;
+import com.rackspacecloud.blueflood.io.*;
+import com.rackspacecloud.blueflood.outputs.formats.MetricData;
+import com.rackspacecloud.blueflood.rollup.Granularity;
+import com.rackspacecloud.blueflood.service.Configuration;
+import com.rackspacecloud.blueflood.service.CoreConfig;
+import com.rackspacecloud.blueflood.types.*;
+import org.apache.avro.generic.GenericData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.xml.transform.Result;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * This class deals with reading/writing metrics to the basic metrics_* and metrics_string column families
+ * using Datastax driver
+ */
+public class DBasicMetricsRW extends DAbstractMetricsRW {
+
+    public static final String COLUMN1 = "column1";
+    public static final String VALUE = "value";
+
+    private static final Logger LOG = LoggerFactory.getLogger( DBasicMetricsRW.class );
+
+    private boolean areStringMetricsDropped = Configuration.getInstance().getBooleanProperty( CoreConfig.STRING_METRICS_DROPPED);
+    private List<String> tenantIdsKept = Configuration.getInstance().getListProperty(CoreConfig.TENANTIDS_TO_KEEP);
+    private Set<String> keptTenantIdsSet = new HashSet<String>(tenantIdsKept);
+
+    private DRawIO rawIO = new DRawIO();
+    private LocatorIO locatorIO = IOContainer.fromConfig().getLocatorIO();
+
+    /**
+     * This method inserts a collection of {@link com.rackspacecloud.blueflood.types.IMetric} objects
+     * to the appropriate Cassandra column family.
+     *
+     * @param metrics
+     *
+     * @throws IOException
+     */
+    @Override
+    public void insertMetrics( Collection<IMetric> metrics ) throws IOException {
+        // if there are strings & booleans in this request, they will be tracked as numeric metrics.
+        // not sure how we want to get around that.
+        Timer.Context ctx = Instrumentation.getWriteTimerContext( CassandraModel.CF_METRICS_FULL_NAME );
+
+        Map<Locator, ResultSetFuture> futures = new HashMap<Locator, ResultSetFuture>();
+
+        try {
+
+            for( IMetric metric : metrics ) {
+
+                if (!shouldPersist(metric)) {
+                    LOG.trace("Metric shouldn't be persisted, skipping insert", metric.getLocator().toString());
```

Comment:
This one doesn't have a format. First argument should be a string format. 

Also, should we be logging INFO? Is that too noisy? If we are dropping metrics, it feels like it should be more than TRACE. But this one is a minor comment. :-) I'm ok either way.